packages/create-react-native-library/templates/native-common/android/build.gradle

```diff
@@ -69,28 +76,62 @@ android {
 <% if (project.arch === "new") { -%>
     buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
 <% } -%>
+<% if (project.cpp || project.nitro) { -%>
 
-<% if (project.cpp) { -%>
     externalNativeBuild {
       cmake {
-        cppFlags "-O2 -frtti -fexceptions -Wall -fstack-protector-all"
+        cppFlags "-frtti -fexceptions -Wall -fstack-protector-all"
+        arguments "-DANDROID_STL=c++_shared"
         abiFilters (*reactNativeArchitectures())
+
+        buildTypes {
+          debug {
+            cppFlags "-O1 -g"
+          }
+          release {
+            cppFlags "-O2"
+          }
+        }
       }
     }
 <% } -%>
   }
-<% if (project.cpp) { -%>
+<% if (project.cpp || project.nitro) { -%>
 
   externalNativeBuild {
     cmake {
       path "CMakeLists.txt"
     }
   }
+
+  packagingOptions {
+    excludes = [
+            "META-INF",
+            "META-INF/**",
+            "**/libc++_shared.so",
+            "**/libfbjni.so",
+            "**/libjsi.so",
+            "**/libfolly_json.so",
+            "**/libfolly_runtime.so",
+            "**/libglog.so",
+            "**/libhermes.so",
+            "**/libhermes-executor-debug.so",
+            "**/libhermes_executor.so",
+            "**/libreactnative.so",
+            "**/libreactnativejni.so",
+            "**/libturbomodulejsijni.so",
+            "**/libreact_nativemodule_core.so",
+            "**/libjscexecutor.so"
+    ]
+  }
```

Comment:
this still need those days?, I created a rn turbo modules recently then migrated to nitro later, and i didn't got any symbol conflict as we used to have in the past

Reply:
I've got these from @mrousavy's nitro template, @mrousavy is it safe to remove these? Also maybe we can make this to default for everyone?

Reply:
I don't know, lol.

Reply:
FYI I tried to build nitro without these, but I got an error mentioning some of these libs were duplicated so these are needed for nitro modules right now.